src/types.rs

```diff
@@ -78,32 +154,33 @@ pub struct Response {
     pub status_code: u16,
     pub response_type: String,
     pub headers: HashMap<String, String>,
-    pub body: String,
+    pub body: ActixBytesWrapper,
     pub file_path: Option<String>,
 }
 
 #[pymethods]
 impl Response {
     #[new]
-    pub fn new(status_code: u16, headers: HashMap<String, String>, body: String) -> Self {
-        Self {
+    pub fn new(status_code: u16, headers: HashMap<String, String>, body: &PyAny) -> PyResult<Self> {
+        return Ok(Self {
             status_code,
             // we should be handling based on headers but works for now
             response_type: "text".to_string(),
             headers,
-            body,
+            body: ActixBytesWrapper::new(body).unwrap(),
```

Comment:
@madhavajay this is the line that makes your test fail. You shouldn't unwrap here as it will make the Rust code panic. Instead, you want the `new` function to return the error. It will then be interpreted to raise an `Exception` in Python, and you will be able to catch it in the test.

```suggestion
            body: ActixBytesWrapper::new(body)?,
```

Reply:
Ah, yes this is much better thank you!
I have added the tests back.